tools/gdal/ogr2ogr.xml

```diff
@@ -2,63 +2,106 @@
-            '$input'
+            #if $input.is_of_type("shp")
+                ${os.path.join( $input.extra_files_path, 'shapefile.shp' )} 
+            #else    
+                '$input'
+            #end if
+            #if $f=='ESRI Shapefile'
+                &&
+                mkdir ${shapefile.extra_files_path} &&
+                mv shapefile.* ${shapefile.extra_files_path}
+            #end if 
     ]]></command>
     <inputs>
-        <param type="data" name="input" format="data" label="Gdal supported input file" help="http://www.gdal.org/ogr_formats.html"/>
-        <param type="select" name="f" label="Conversion format" value=""
-            help="Select the format to convert your input file. Currently supported : Shapefile, GeoJSON, GML, KML and WFS. The shapefiles outputs shp, shx, dbf and prj will populate a datacollection.">
-            <option value="ESRI Shapefile">ESRI Shapefile</option> 
+        <param type="data" name="input" format="shp,xml,json,txt" label="Gdal supported input file" help="http://www.gdal.org/ogr_formats.html"/>
+        <param type="select" name="f" label="Conversion format" value="" help="Select the format to convert your input file. Currently supported : Shapefile, GeoJSON, GML, KML and WFS.">
+            <option value="ESRI Shapefile">ESRI Shapefile</option>
             <option value="GeoJSON">GEOJSON</option> 
             <option value="GML">GML</option>
-            <option value="KML">KML</option>
+	    <option value="KML">KML</option>
             <option value="WFS">WFS</option>
         </param>
     </inputs>
     <outputs>
-        <data name="output" format="xml" label="${on_string}.${f}">
+        <data name="output" from_work_dir="output" format="xml" label="${on_string}.${f}">
             <filter>f != 'ESRI Shapefile'</filter>
                 <change_format>
-                    <when input="f" value="GeoJson" format="json"/> 
+                    <when input="f" value="GeoJSON" format="json"/> 
                     <when input="f" value="GML" format="xml"/> 
                     <when input="f" value="KML" format="xml"/> 
                     <when input="f" value="WFS" format="txt"/> 
                 </change_format> 
         </data>
-        <collection name="shapefile" type="list" label="${on_string}.shapefile">
+        <data name="shapefile" format="shp" from_work_dir="shapefile" label="${on_string}.shapefile">
             <filter>f == 'ESRI Shapefile'</filter>
-                <data name="shp" format="binary" from_work_dir="*.shp"/>
-                <data name="shx" format="binary" from_work_dir="*.shx"/>
-                <data name="dbf" format="binary" from_work_dir="*.dbf"/>
-                <data name="prj" format="txt" from_work_dir="*.prj"/>
-       </collection>
+        </data>
     </outputs>
-    
     <tests>
-        <test> <!-- ogr2ogr -f KML custom.geo.kml custom.geo.json -->
-            <param name="input" value="custom.geo.json"/>
+        <test>
```

Comment:
You could use that trick: https://github.com/galaxyproject/tools-iuc/pull/2444/files#diff-182bc3778630f0a87cb6db71e73edf6aR380

And test your number of outputs and with this your filters.